src/moment_kinetics.jl

```diff
@@ -416,10 +326,41 @@ function setup_moment_kinetics(input_dict::Dict; restart_prefix_iblock=nothing,
     else
         restarting = true
 
+        if restart === true
+            run_name = input_dict["run_name"]
+            base_directory = get(input_dict, "base_directory", "runs")
+            output_dir = joinpath(base_directory, run_name)
+            io_settings = get(input_dict, "output", Dict{String,Any}())
+            binary_format = get(io_settings, "binary_format", hdf5)
+            if binary_format === hdf5
+                ext = "h5"
+            elseif binary_format === netcdf
+                ext = "cdf"
+            else
+                error("Unrecognized binary_format '$binary_format'")
+            end
+            restart = glob(joinpath(output_dir, run_name * ".dfns*." * ext))[1]
```

Comment:
This line appears to be giving BoundsError when the number of shared memory regions is greater that 1.

Reply:
Does this code make any allowances for multiple .h5 or .cdf files to be copied and read? I cannot see how it does here.

Reply:
Trying 

println(joinpath(output_dir, run_name * ".dfns*." * ext))
println(glob(joinpath(output_dir, run_name * ".dfns*." * ext)))

Gets 

runs/wall-bc_cheb_single_ion/wall-bc_cheb_single_ion.dfns*.h5
String[]

It's pretty clear that this code is broken for some reason in this case. What should it be doing? What does the function `glob' do? I cannot find the definition.

Reply:
`glob()` finds files matching a pattern (name comes from *nix shell, e.g. https://tldp.org/LDP/abs/html/globbingref.html). The bounds error usually means there is no file matching the pattern it is looking for.

Some of the error messages can be a bit unhelpful. I tried to improve things a bit in a few commits in #128. Those would need merging if we merge this PR first - maybe I should just cherry-pick them into this branch...